SocketIOClientSwift/SocketParser.swift

```diff
@@ -61,169 +60,119 @@ class SocketParser {
         }
     }
     
-    private static func handleEvent(p: SocketPacket, socket: SocketIOClient) {
-        if !isCorrectNamespace(p.nsp, socket) {
-            return
-        }
-        
-        socket.handleEvent(p.event, data: p.args,
-            isInternalMessage: false, wantsAck: p.id)
-    }
-    
     // Translation of socket.io-client#decodeString
-    static func parseString(str: String) -> SocketPacket? {
-        let arr = Array(str.characters)
-        let type = String(arr[0])
+    static func parseString(str: String) throws -> SocketPacket {
```

Comment:
I would much prefer that this be kept an optional return type. Or if Swift ever introduces an Either type, that would be better